contracts/v0.8/utils/FilAddresses.sol

```diff
@@ -68,20 +74,20 @@ library FilAddresses {
     }
 
     /// @notice allow to validate if an address is valid or not
-    /// @dev we are only validating known address types. If the type is not known, the default value is true
+    /// @dev we are only validating known address types. If the type is not known, the default value is false
     /// @param addr the filecoin address to validate
     /// @return whether the address is valid or not
     function validate(CommonTypes.FilAddress memory addr) internal pure returns (bool) {
         if (addr.data[0] == 0x00) {
-            return addr.data.length <= 10;
+            return (addr.data.length > 1 && addr.data.length <= 11);
         } else if (addr.data[0] == 0x01 || addr.data[0] == 0x02) {
             return addr.data.length == 21;
         } else if (addr.data[0] == 0x03) {
             return addr.data.length == 49;
-        } else if (addr.data[0] == 0x04) {
-            return addr.data.length <= 64;
+        } else if (addr.data[0] == 0x04 && addr.data[1] == 0x0a) {
+            return addr.data.length == 22;
         }
 
-        return addr.data.length <= 256;
```

Comment:
Why are we changing the default result? This is in conflict with the comment above that `we are only validating known address types`

Reply:
If the type is unknown or invalid, we return `false`. Leaving it as it is would result in `true` for unknown or invalid types. In the future, if new types are introduced, this function will need modification to accommodate them. Currently, it supports all known addresses types 0/1/2/3/4(EAM).

Reply:
Yes I understand. I was just wondering whether we want to keep this compatible with future address types (f5 etc) that we don't know about yet. Right now, we aren't able to distinguish between eg a) invalid known address eg F1 and b) unknown address eg F5.
My suggestion for now would be to stick with current handling of unknown addresses. I'm not 100% clear what the rationale was but I'm not comfortable with changing it. Wdyt?

Reply:
Hm, I'm a bit afraid of reverts because the calling contract won't be able to handle them, right?

Reply:
It might be a bit tricky. Let's retain the previous validate implementation with default true value and with length changes in the if statements. We should be aware that this function may return 'true' for some invalid formats, and we can consider changing it if necessary later. Is that OK?